### PR TITLE
(Py2) Fix DFV queries

### PR DESCRIPTION
**Pull request by rod-glover**
This PR fixes incorrect `DataFileVariable` queries (for `modelmeta` >= 0.3.0, which is now our target version).

Note that this PR merges to branch `py2`.

This PR includes the following items:
- Fix for queries
- Fix for broken (bit rot) Python CI build
- Bump to version 1.2.1. A bit premature, but that was the best way to get a test build of PDP working.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -298,14 +315,19 @@ def variable_aliases():
 
 
 @pytest.fixture(scope="function")
-def dfv_dsg_tss(data_files, variable_aliases):
+def grids():
+    return make(make_grid, 1)
+
+
+@pytest.fixture(scope="function")
+def dfv_dsg_tss(data_files, variable_aliases, grids):
```
Comment (jameshiebert) [inline tests/conftest.py:323]: Maybe just pick out the first (only) grid here?

```python
grid = next(grids)
return make(
    make_dfv_gridded
    [
         (data_files[0], variable_aliases[0], grid),
...
```

**Review by jameshiebert — COMMENTED** [review]
Looks pretty straightforward. 80% of the changes are just formatting/bitrot, but I have a few super minor comments.

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
 # News / Release Notes
 
+## 1.2.1
+
+*Release Date: 2021-Jun-14*
+
+Release for integration test with `pdp`. 
+
+- Fixes DataFileVariable queries for modelmeta >=0.3.0
```
[the remark this answers is not in the document]
Comment (rod-glover) [reply NEWS.md:9]: Yes, regrettably. I updated them quite a long time ago ... wrongly. Minor change, but yeesh.

File: .github/workflows/python-ci.yml [hunk 3]
```diff
@@ -20,6 +20,10 @@ jobs:
       env:
         PIP_INDEX_URL: https://pypi.pacificclimate.org/simple
       run: |
+        sudo apt-get install -y curl ca-certificates gnupg
```
[the remark this answers is not in the document]
Comment (rod-glover) [reply .github/workflows/python-ci.yml:23]: I think that might be an excellent idea. All the builds -- prod images, local test images, Python CI -- I think depend on the same basis and we could use a single common custom base image. How about if I do that in a separate PR?

**Pull request merged**